src/wazuh_modules/wm_exec.c

```diff
@@ -346,8 +346,10 @@ int wm_exec(char *command, char **output, int *exitcode, int secs, const char *
             }
 
             char *new_env = getenv("PATH");
-            mdebug1("New 'PATH' environment variable set: '%s'", new_env);
-            free(new_path);
+            if (new_env != NULL) {
+                mdebug1("New 'PATH' environment variable set: '%s'", new_env);
+                free(new_path);
```

Comment:
`new_path` must always be freed since it is allocated at the start of the function. In addition, do the same change in the Windows version of the function for consistency.